src/go/rpk/pkg/cli/registry/subject.go

```diff
@@ -0,0 +1,144 @@
+	var deleted bool
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "Display all subjects",
+		Args:    cobra.ExactArgs(0),
+		Run: func(*cobra.Command, []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]string{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewSRClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			subjects, err := cl.Subjects(context.Background(), sr.HideShowDeleted(deleted))
+			out.MaybeDieErr(err)
+
+			if isText, _, s, err := f.Format(subjects); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
+			for _, s := range subjects {
+				fmt.Println(s)
+			}
+		},
+	}
+	cmd.Flags().BoolVar(&deleted, "deleted", false, "If true, list deleted subjects as well")
+	return cmd
+}
+
+type deleteResponse struct {
+	Subject  string `json:"subject" yaml:"subject"`
+	Versions []int  `json:"versions,omitempty" yaml:"versions,omitempty"`
+	Err      string `json:"error,omitempty" yaml:"error,omitempty"`
+}
+
+func subjectDeleteCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var permanent bool
+	cmd := &cobra.Command{
+		Use:   "delete [SUBJECT...]",
+		Short: "Soft or hard delete subjects",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(_ *cobra.Command, subjects []string) {
+			f := p.Formatter
+			if h, ok := f.Help([]deleteResponse{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewSRClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			var (
+				wg      sync.WaitGroup
+				mu      sync.Mutex
+				results []deleteResponse
+			)
+
+			for i := range subjects {
+				subject := subjects[i]
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					versions, err := cl.DeleteSubject(context.Background(), subject, sr.DeleteHow(permanent))
+					mu.Lock()
+					defer mu.Unlock()
+					var errStr string
+					if err != nil {
+						errStr = err.Error()
+					}
+					results = append(results, deleteResponse{
+						subject,
+						versions,
+						errStr,
+					})
+				}()
+			}
+			wg.Wait()
+			if isText, _, s, err := f.Format(results); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
+			tw := out.NewTable("subject", "versions-deleted", "error")
+			defer tw.Flush()
+			for _, r := range results {
+				tw.PrintStructFields(struct {
+					Subject  string
+					Versions []int
+					Err      string
+				}{
+					r.Subject,
+					r.Versions,
+					r.Err,
+				})
+			}
+		},
+	}
+	cmd.Flags().BoolVar(&permanent, "permanent", false, "Perform a hard (permanent) delete of the subject; requires a soft-delete first")
+	return cmd
```

Comment:
There's not a force option to get it done in one?

Reply:
Is this per the SR specification @twmb ?

Reply:
Schema's _must_ be soft deleted before they can be hard deleted. We could potentially do two deletes inside the function itself, though, rather than requiring an end user to do the same command twice (the second time with an extra flag)

Reply:
That's pretty much all I'd be looking for from an admin perspective

Reply:
So `--force-permanent`? And we issue the 2 requests and indicate in the error message if we got an error in the soft or permanent deletion?

Reply:
Yeah pretty much

Reply:
Also fine from my perspective

Reply:
Added `--force` flag to both `rpk registry subject delete` and `rpk registry schema delete`. Thanks!